src/regex.nim

```diff
@@ -2100,6 +2101,51 @@ proc groupsCount*(m: RegexMatch): int =
   ##
   m.groups.len
 
+
+proc groupsNames*(m: RegexMatch): seq[string] =
+  ## return the names of capturing groups
+  ##
+  ## .. code-block:: nim
+  ##   block:
+  ##     var m: RegexMatch
+  ##     doAssert "hello world".match("(?P<greet>hello) (?P<who>world)", m)
+  ##     doAssert m.groupsCount == 2
+
+
+  toSeq(m.namedGroups.keys)
+
+
+proc groupsByName*(m: RegexMatch, name: string, originalText: string): seq[string] =
+  ## return capturing groups by name
+  ##
+  ## .. code-block:: nim
+  ##   block:
+  ##     var m: RegexMatch
+  ##     doAssert "hello world".match("(?P<greet>hello) (?P<who>world)", m)
+  ##     doAssert m.groupByName("greet") == @["hello"]
+  ##     doAssert m.groupByName("who") == @["world"]
+  
+  for bounds in m.group(name):
+    result.add(originalText[bounds])
```

Comment:
`result` needs explicit initialization in Nim 0.18 (`newSet[string]()`), I think that's why CI is failing.